ddtrace/tracer/sampler.go

```diff
@@ -188,19 +188,76 @@ func newRulesSampler(rules []SamplingRule) *rulesSampler {
 	}
 }
 
-// samplingRulesFromEnv parses sampling rules from the DD_TRACE_SAMPLING_RULES
-// environment variable.
+// samplingRulesFromEnv parses sampling rules from the DD_TRACE_SAMPLING_RULES environment variable
+// and combines them with spanSamplingRules
```

Comment:
It's atypical that a Go function returns both a non-nil value and a non-nil error. That should be documented here. e.g. "// This function may return a non-nil error as well as a non-nil list of sampling rules that were successfully parsed."

Reply:
It's a bit confusing. Can we just have one single function called `samplingRulesFromEnv` instead of 3 functions (two which get sampling rules from separate envs and one which merges them).

Reply:
Oh, this is only used in tests, and in `log.go`. I think it would simplify things a lot if you just deleted it and instead called both in `log.go` and test them individually (or create a helper method in the tests instead).

Reply:
If what you're saying is "delete three functions and combine the values into one variable as it used to be", it won't work. We need them in separate variables, so that we don't run single span rules unnecessarily before we know the trace sampling decision is drop. I can combine them into 1 function, and pass config in there though

Reply:
It's atypical, but it's how it was before I had any changes as well 😅

Reply:
I just meant to delete the one used _only_ in tests and `log.go` and keep the other two. Ignore the first comment :)

Reply:
Noted, removed